### PR TITLE
Add address canary check for routers and brokers

### PR DESCRIPTION
**Pull request by lulf**
Add a periodic health check for standard address space, where pre-configured canary addresses are added to the routers and brokers. The periodic check will perform a send and receive to the canary addresses, and report prometheus metrics with the result.
    
An example alert is provided that alerts on failed canaries for more than 5 minutes.

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests profile=shared

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated

File: standard-controller/src/main/java/io/enmasse/controller/standard/AddressCanaryHealth.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    @Override
+    public void run() {
+        while (running) {
+            try {
+                Thread.sleep(checkInterval.toMillis());
+                checkHealth(kubernetes.listClusters());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```
Comment (k-wall) [inline standard-controller/src/main/java/io/enmasse/controller/standard/AddressCanaryHealth.java:128]: either break the loop, or set running false on interrupt.

File: standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    public Set<String> probeAddresses(Set<String> addresses, Duration timeout) {
+        if (connection == null) {
+            throw new RuntimeException("HelathProbeClient not connected");
+        }
+
+        String payload = String.format("PING %s", UUID.randomUUID().toString());
+        Message message = Proton.message();
+        message.setBody(new AmqpValue(payload));
```
Comment (k-wall) [inline standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java:79]: the probe queue is a LVQ, don't we also need to set a LVQ property in order for the conflation to take place? https://activemq.apache.org/components/artemis/documentation/latest/last-value-queues.html

File: standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        }
+        return new HashSet<>(passed);
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (context != null) {
+            CompletableFuture.allOf(
+                    runOnContext(context, () -> {
+                        for (ProtonSender sender : senders) {
```
Comment (k-wall) [inline standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java:145]: Missing synchronisation

File: standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    public void connect(String host, int port, ProtonClientOptions clientOptions, CompletableFuture<Void> promise) {
+        if (connection != null) {
+            log.debug("Already connected");
+            promise.complete(null);
+            return;
+        }
+
+        ProtonClient client = ProtonClient.create(vertx);
+        log.debug("Connecting to {}:{}", host, port);
+        client.connect(clientOptions, host, port, result -> {
```
Comment (k-wall) [inline standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java:57]: Perhaps add a connection properties to help identity the work of the heath check in the router log.

Comment (lulf) [reply standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java:57]: I believe the containerId is set to `standard-controller-healthcheck`. Is this sufficient, or should it be set in properties as well?

File: standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java (hunk 2 above)
Comment (lulf) [reply standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java:83]: Fixed.

**Comment by lulf** [comment]
@enmasse-ci run tests profile=acceptance

File: standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java (hunk 4 above)
Comment (k-wall) [reply standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java:60]: Container id is sufficient.

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=MonitoringTest

**Pull request merged**